src/main/java/uk/gov/hmcts/divorce/caseworker/service/EmailUpdateService.java

```diff
@@ -20,29 +28,45 @@ public class EmailUpdateService {
     InviteApplicantToCaseNotification inviteApplicantToCaseNotification;
     @Autowired
     NotificationDispatcher notificationDispatcher;
+    @Autowired
+    CcdAccessService ccdAccessService;
 
     public CaseDetails<CaseData, State> processEmailUpdate(final CaseDetails<CaseData, State> caseDetails,
                                                       final CaseDetails<CaseData, State> beforeCaseDetails,
                                                       boolean isApplicant1) {
 
         final CaseData data = caseDetails.getData();
+        final CaseData beforeData = beforeCaseDetails.getData();
 
-        Applicant applicant = isApplicant1 ? data.getApplicant1() : data.getApplicant2();
+        final Applicant applicant = isApplicant1 ? data.getApplicant1() : data.getApplicant2();
+        final Applicant partner = isApplicant1 ? data.getApplicant2() : data.getApplicant1();
+        final Applicant beforeApplicant = isApplicant1 ? beforeData.getApplicant1() : beforeData.getApplicant2();
 
-        if (applicant.getEmail() == null || applicant.getEmail().isBlank()
-            || applicant.isRepresented()) {
-            return caseDetails;
-        }
+        if (isEmailBeingRemoved(beforeApplicant, applicant)) {
+            if (doesApplicantNeedToBeMadeOffline(data,isApplicant1)) {
+                log.info("Party will be made offline and any case access removed for Case Id: {}", caseDetails.getId());
+                final var roles = (data.getApplicationType() == ApplicationType.SOLE_APPLICATION)
+                    ? isApplicant1 ? List.of(CREATOR.getRole()) : List.of(APPLICANT_2.getRole())
+                    : List.of(CREATOR.getRole(), APPLICANT_2.getRole());
 
-        //Do not send invite to respondent if sole application and case hasn't been issued
-        if (data.getApplicationType().isSole() && (data.getApplication().getIssueDate() == null) && !isApplicant1) {
-            return  caseDetails;
+                applicant.setOffline(YesOrNo.YES);
+                if (data.getApplicationType() == ApplicationType.JOINT_APPLICATION) {
+                    partner.setOffline(YesOrNo.YES);
+                }
+                ccdAccessService.removeUsersWithRole(caseDetails.getId(), roles);
+            }
+        } else {
+            if (applicant.getEmail() == null || applicant.getEmail().isBlank()
+                || applicant.isRepresented()) {
+                return caseDetails;
+            }
+            //Do not send invite to respondent if sole application and case hasn't been issued
+            if (data.getApplicationType().isSole() && (data.getApplication().getIssueDate() == null) && !isApplicant1) {
+                return caseDetails;
+            }
```

Comment:
You wouldn't need the comment with a variable:
```suggestion
            boolean isRespondentAndSoleCaseNotIssued = data.getApplicationType().isSole() && data.getApplication().getIssueDate() == null && !isApplicant1;
            if (isRespondentAndSoleCaseNotIssued) {
                return caseDetails;
            }
```